### PR TITLE
CT-1418 Select Case Type interstitial page

### PR DESCRIPTION
**Pull request by stephenrichards**
This feature is switched on/off with the FeatureSet settings.

### CODE REVIEW

File: app/views/cases/_new_form_case_type.html.slim [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/ - if params[:case][:type] == 'Case::FOI'
+= f.radio_button_fieldset :type,
+            choices: [Case::FOI,
+                      Case::FOI::TimelinessReview,
+                      Case::FOI::ComplianceReview]
+/ - else
+/   = f.hidden_field :type
```
Comment (misaka) [inline app/views/cases/_new_form_case_type.html.slim]: Do we still need the commented-out bits here?

Comment (beccapearce) [reply app/views/cases/_new_form_case_type.html.slim]: I think we need them in but not commented out, it's so we don't ask about internal reviews on when we make a SAR

**Review by misaka — CHANGES_REQUESTED** [review]

File: app/views/cases/_new_form_case_type.html.slim (hunk 1 above)
Comment (misaka) [reply app/views/cases/_new_form_case_type.html.slim]: I'm going to leave this as-is and sort this out as part of the next ticket, CT-1391.

**Review by misaka — APPROVED** [review]

File: app/controllers/cases_controller.rb [hunk 2]
```diff
@@ -109,10 +109,19 @@ def filter
     redirect_to redirect_url
   end
 
+  def select_type
+    @case = Case.new
+  end
+
   def new
+    if FeatureSet.sars.enabled?
+      @type = params[:case][:type]
```
Comment (beccapearce) [inline app/controllers/cases_controller.rb]: We're still not using the type in the case creation, we're just making a standard Case

File: app/controllers/cases_controller.rb [hunk 3]
```diff
@@ -109,10 +109,19 @@ def filter
...
+    end
+
+
     authorize Case.new(category: Category.foi), :can_add_case?
 
-    @case = Case.new(type: nil)
+    @case = Case.new
```
Comment (beccapearce) [inline app/controllers/cases_controller.rb]: Is it worth adding a rescue if no case type is set?

File: app/controllers/cases_controller.rb (hunk 2 above)
Comment (misaka) [reply app/controllers/cases_controller.rb]: Ya, there's a separate ticket to deal with case creation ... CT-1391, I think that using the case type can go there.

**Review by beccapearce — APPROVED** [review]

File: app/controllers/cases_controller.rb (hunk 3 above)
Comment (misaka) [reply app/controllers/cases_controller.rb]: Sounds like it should raise an error to the user. As above, I think that we can handle that in CT-1391.

**Pull request merged**